### PR TITLE
Memory leaks & uninitialized value fixes

### PR DESCRIPTION
**Pull request by bkueng**
Some fixes, found with Clang static analysis.

### CODE REVIEW

File: src/modules/position_estimator_inav/position_estimator_inav_main.cpp [hunk 1]
```diff
@@ -410,7 +410,7 @@ int position_estimator_inav_thread_main(int argc, char *argv[])
 
 	/* wait for initial baro value */
 	bool wait_baro = true;
-	TerrainEstimator *terrain_estimator = new TerrainEstimator();
```
Comment (julianoes) [inline src/modules/position_estimator_inav/position_estimator_inav_main.cpp]: This change probably needs a raise in stack size.

Comment (bkueng) [inline src/modules/position_estimator_inav/position_estimator_inav_main.cpp]: You think so? Look at TerrainEstimator, it contains only some floats and timestamps, <100B.

Comment (julianoes) [inline src/modules/position_estimator_inav/position_estimator_inav_main.cpp]: I didn't look at the code, but I'd suggest to just run it and look at `top`.

Comment (bkueng) [inline src/modules/position_estimator_inav/position_estimator_inav_main.cpp]: It's constant at 4929/ 5296, so we're still fine.

**Pull request merged**